remote/bitbucketserver/internal/client.go

```diff
@@ -165,6 +174,22 @@ func (c *Client) DeleteHook(owner string, name string, link string) error {
 	return c.doPut(fmt.Sprintf(pathHookEnabled, c.base, owner, name, hookName), hookBytes)
 }
 
+func (c *Client) GetHookDetails(owner string, name string) (*HookPluginDetails, error) {
+	urlString := fmt.Sprintf(pathHookDetails, c.base, owner, name, hookName)
+	response, err := c.client.Get(urlString)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+	contents, err := ioutil.ReadAll(response.Body)
+	hookDetails := HookPluginDetails{}
+	err = json.Unmarshal(contents, &hookDetails)
+	if err != nil {
+		return nil, err
+	}
+	return &hookDetails, nil
```

Comment:
in this case I would `return &hookDetails, err`. The calling function will generally check` err != nil` so there is no harm returning both